src/modules/clustering/hierarchical_clustering_method.cpp

```diff
@@ -104,6 +106,22 @@ int junction_distance(Junction const & lhs, Junction const & rhs)
     }
 }
 
+std::vector<Junction> subsample_partition(std::vector<Junction> const & partition, uint16_t const sample_size)
+{
```

Comment:
I would throw in a `assert(partition.size() >= sample_size)` to indicate that a sample_size has to be larger or this algorithm doesn't work.

Reply:
I'm wondering if this assertion needs to be here... The only time the subsample is called is within the if statement below
```cpp
if (partition_size > max_partition_size)
    // ...
    partition = subsample_partition(partition, max_partition_size);
```
So it seems a bit redundant to check it first in the if statement and then again within the function itself?

Reply:
True, but the `assert` will not be part of the program if it is build in Release.
I see it as a good way to document precondition that have to be hold met.

Reply:
> True, but the `assert` will not be part of the program if it is build in Release.
> I see it as a good way to document precondition that have to be hold met.

I agree because I see each function as kind of a module that can be used and re-used multiple times. The `assert` is helpful to make sure that it is used in a correct manner.